<launch>
	<!-- Camera parameters -->
	<param name="set_gain" value="40" />
	<param name="set_exposure" value="6000" />
	<param name="set_height" value="240" />
	<param name="set_width" value="320" />
	<param name="set_bright" value="10" />
	<node name="ASIReadcolor" pkg="read_web_cam" type="ASIReadcolor"/>

	<node name="color_tracking_node" pkg="ukftest" type="color_tracking_node"/> 		

	<node name="urg_node" pkg="urg_node" type="urg_node" />

	<node name="my_vfh_v2_node" pkg="ukftest" type="my_vfh_v2_node" />

	<node name="serial_to_uav_node" pkg="serial_to_uav" type="serial_to_uav_node"  />
	<param name="is_write_on" value="true" />
	
	<node name="neg_estimation_node" pkg="ukftest" type="neg_estimation_node" />

	<node name="uav_avoid_neg_controller_node" pkg="ukftest" type="uav_avoid_neg_controller_node" />
	<!-- Avoiding PID parameters -->

	<param name="Kp_avoid_vel" value="2.7" />
	<param name="Ki_avoid_vel" value="0.0" />
	<param name="Kd_avoid_vel" value="0.2" />
	<param name="pid_gain_avoid" value="1.0" />
	<param name="controlLimit_avoid" value="30.0" />

	
	<node name="uav_track_neg_controller_node" pkg="ukftest" type="uav_track_neg_controller_node" />
	<!-- Tracking PID parameters -->
	<param name="Kp_pos" value="1.0" />
	<param name="Ki_pos" value="0.0" />
	<param name="Kd_pos" value="0.0" />
	<param name="Kp_vel" value="1.20" />
	<param name="Ki_vel" value="0.20" />
	<param name="Kd_vel" value="0.34" />
	<param name="pid_gain_track" value="1.0" />
	<param name="controlLimit_track" value="3.0" />

	<!-- <node name="uav_random_neg_controller_node" pkg="ukftest" type="uav_random_neg_controller_node" /> -->
	<node name="uav_random_local_controller_node" pkg="ukftest" type="uav_random_local_controller_node" />
	<!-- Random-flying parameters -->
	<param name="loop_time" value="9.0" />
	<param name="random_flying_time" value="4.0" />
	<param name="init_pitch_value" value="0.0" />
	<param name="init_roll_value" value="0.0" />
	<param name="random_pitch_range" value="100.0" />
	<param name="random_roll_range" value="150.0" />
	<param name="random_x_0" value="1.0" />
	<param name="random_y_0" value="5.0" />
	<param name="random_x_1" value="4.0" />
	<param name="random_y_1" value="16.0" />


	<node name="uav_boundary_local_controller_node" pkg="ukftest" type="uav_boundary_local_controller_node" />
	<param name="pitch_range" value="150.0" />
	<param name="roll_range" value="150.0" />
	<param name="x_0" value="1.0" />
	<param name="y_0" value="5.0" />
	<param name="x_1" value="4.0" />
	<param name="y_1" value="16.0" />
	<param name="x_2" value="-10.0" />
	<param name="y_2" value="-10.0" />
	<param name="x_3" value="-10.0" />
	<param name="y_3" value="-10.0" />


	<node name="uav_takeoff2_controller_node" pkg="ukftest" type="uav_takeoff2_controller_node"  />
	<!-- Landing parameters 20min-->
	<param name="takeoff_time" value="15.0" />
	<param name="set_height" value="1.7" />

	<!--<node name="uav_landing_neg_controller_node" pkg="ukftest" type="uav_landing_neg_controller_node" /> -->
	<!-- Landing parameters 20min-->
	<param name="landing_time" value="12000.0" />
	
	<node name="uav_state2_controller_node" pkg="ukftest" type="uav_state2_controller_node" output="screen" />
	<!-- status parameters -->
	<param name="yaw_rate" value="0.0" />
	

</launch>

